### PR TITLE
Add support for Rails 8.0

### PR DESCRIPTION
**Pull request by phlipper**
# Overview

This PR adds support for Rails 8.0 which has been officially released: https://rubyonrails.org/2024/11/7/rails-8-no-paas-required. 

## Notes

* Update `Appraisals`
* Update CI Workflow
* Add a new CHANGELOG entry

Thank you for all of your great work on this gem!

### CODE REVIEW

**Review by Spone — APPROVED** [review]
Thanks!

**Comment by phlipper** [comment]
The CI workflow has been updated to support Rails 8 properly. Sorry about that!

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -24,6 +24,7 @@ jobs:
           - { ruby: "3.1", rails: "7.2", rubygems: "default" }
           - { ruby: "3.2", rails: "7.2", rubygems: "default" }
           - { ruby: "3.3", rails: "7.2", rubygems: "default" }
+          - { ruby: "3.3", rails: "8.0", rubygems: "default" }
           - { ruby: "3.2", rails: "head", rubygems: "latest" }
```
[the remark this answers is not in the document]
Comment (Spone) [reply .github/workflows/main.yml:28]:
```suggestion
```

[the remark this answers is not in the document]
Comment (phlipper) [reply .github/workflows/main.yml]: @Spone, thanks for catching that! I have pushed an update to remove Ruby 3.2 for Rails `head` in CI.

**Comment by Spone** [comment]
Awesome, thanks for your PR @phlipper :)

**Comment by bilby91** [comment]
Is there an ETA for releasing a new version with this change ?

Thanks!

**Comment by nicolas-brousse** [comment]
[v0.2.9](https://github.com/pantographe/view_component-form/releases/tag/v0.2.9) released :blush:

**Pull request merged**